### PR TITLE
detecting services and infer languages

### PR DESCRIPTION
**Pull request by DanielMcSheehy**
added installation instructions for how to disable language detection:

Optionally you may also pass in a list of features you want to disable, such as
Language Detection.

```bash
-e DISABLE="LANGUAGE_DETECTION"
```

{"level":"info","ts":1614788140.4459164,"caller":"github/integration.go:140","msg":"processing repository","repository":"https://github.com/effxhq/cluster-state.git"}
2021/03/03 16:15:42 Successfully detected  9  service(s)

{"level":"info","ts":1614788099.0079072,"caller":"github/integration.go:140","msg":"processing repository","repository":"https://github.com/effxhq/backeffx.git"}
go   1.13

### CODE REVIEW

File: internal/controller/config.go [hunk 1]
```diff
@@ -46,6 +48,13 @@ func DefaultConfigWithFlags() (*Configuration, []cli.Flag) {
 			Value:       cfg.Workers,
 			EnvVars:     []string{"WORKERS"},
 		},
+		&cli.BoolFlag{
+			Name:        "disable language and version detection",
```
Comment (mjpitz) [inline internal/controller/config.go:52]: this should just be "disable"

File: internal/controller/config.go [hunk 2]
```diff
@@ -46,6 +48,13 @@ func DefaultConfigWithFlags() (*Configuration, []cli.Flag) {
 			Value:       cfg.Workers,
 			EnvVars:     []string{"WORKERS"},
 		},
+		&cli.BoolFlag{
```
Comment (mjpitz) [inline internal/controller/config.go:51]: We probably want this to be more granular then on / off. (see chat)

File: internal/effx/config.go [hunk 3]
```diff
@@ -43,6 +44,13 @@ func DefaultConfigWithFlags() (*Configuration, []cli.Flag) {
 			Value:       cfg.APIKey,
 			EnvVars:     []string{"EFFX_API_KEY"},
 		},
+		&cli.StringFlag{
```
Comment (mjpitz) [inline internal/effx/config.go:47]: I'd look at `cli.StringSliceFlag` which will handle repeated values for you.

Comment (DanielMcSheehy) [reply internal/effx/config.go]: thanks, that makes it a little easier. done

File: internal/run/consumer.go [hunk 4]
```diff
@@ -141,19 +142,35 @@ func (c *Consumer) Consume(log *zap.Logger, repository *model.Repository) (err e
 		// set common annotations for this yaml
 		tags := cloneMap(repository.Tags)
 		annotations := cloneMap(repository.Annotations)
+		inferredTags := []string{}
+
+		if !c.EffxClient.IsFeatureDisabled("LANGUAGE_DETECTION") {
```
Comment (mjpitz) [inline internal/run/consumer.go:147]: We probably want to define this "LANGUAGE_DETECTION" string as a const somewhere. Maybe a `features.go` file in the same directory?

**Review by mjpitz — APPROVED** [review]

File: internal/run/consumer.go (hunk 4 above)
Comment (DanielMcSheehy) [reply internal/run/consumer.go:147]: good call I like that!

**Pull request merged**